File: lib/features/auth/models/student.dart
import 'dart:convert';

class Student {
  String id;
  String firstName;
  String lastName;
  String email;
  String username;
  String password;
  String specialization;
  String year;
  String semester;
  String group;
  String token;

  Student({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.username,
    required this.password,
    required this.specialization,
    required this.year,
    required this.semester,
    required this.group,
    required this.token,
  });

  Student.update({
    required this.id,
    this.firstName = '',
    this.lastName = '',
    this.email = '',
    this.username = '',
    this.password = '',
    this.specialization = '',
    this.year = '',
    this.semester = '',
    this.group = '',
    this.token = '',
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'username': username,
      'password': password,
      'specialization': specialization,
      'year': year,
      'semester': semester,
      'group': group,
      'token': token,
    };
  }

  factory Student.fromMap(Map<String, dynamic> map) {
    return Student(
      id: map['_id'],
      firstName: map['firstName'],
      lastName: map['lastName'],
      email: map['email'],
      username: map['username'],
      password: map['password'],
      specialization: map['specialization'],
      year: map['year'],
      semester: map['semester'],
      group: map['group'],
      token: map['token'],
    );
  }

  String toJson() => json.encode(toMap());

  factory Student.fromJson(String source) =>
      Student.fromMap(json.decode(source) as Map<String, dynamic>);
}

File: server/models/student.js
const mongoose = require('mongoose');

const studentSchema = mongoose.Schema({
    firstName: {
        required: true,
        type: String,
        trim: true,
    },
    lastName: {
        required: true,
        type: String,
        trim: true,
    },
    email: {
        required: true,
        type: String,
        trim: true,
        unique: true
    },
    username: {   
        required: true,
        type: String,
        trim: true,
        unique: true 
    },
    password: {
        required: true,
        type: String,
    },
    specialization: {
        required: true,
        type: String,
    },
    year: {
        required: true,
        type: String
    },
    semester: {
        required: true,
        type: String
    },
    group: {    // Changed from 'section' to 'group'
        required: true,
        type: String
    }
})

const Student = mongoose.model("students", studentSchema);
module.exports = Student;
